### PR TITLE
Feature/webops 243 readiness liveness

### PR DESCRIPTION
**Pull request by singularo**
Add new fields for the liveness and readiness probes.

### CODE REVIEW

File: web/modules/custom/shepherd/shp_content_types/shp_content_types.module [hunk 1]
```diff
@@ -94,3 +95,19 @@ function shp_content_types_node_delete(NodeInterface $node) {
     \Drupal::service('shp_content_types.group_manager')->delete($node);
   }
 }
+
+/**
+ * Implements hook_form_alter().
+ */
+function shp_content_types_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form['field_shp_liveness_probe_params']['#states'] = [
```
Comment (pingers) [inline web/modules/custom/shepherd/shp_content_types/shp_content_types.module]: Lol, I think you missed an if statement here :)

Comment (pingers) [reply web/modules/custom/shepherd/shp_content_types/shp_content_types.module]: Or implement hook_FORM_ID_form_alter().

File: web/modules/custom/shepherd/shp_content_types/config/install/core.entity_form_display.node.shp_project.default.yml [hunk 2]
```diff
@@ -6,10 +6,47 @@ dependencies:
     - field.field.node.shp_project.field_shp_builder_image
     - field.field.node.shp_project.field_shp_env_vars
     - field.field.node.shp_project.field_shp_git_repository
+    - field.field.node.shp_project.field_shp_liveness_probe_params
+    - field.field.node.shp_project.field_shp_liveness_probe_port
+    - field.field.node.shp_project.field_shp_liveness_probe_type
+    - field.field.node.shp_project.field_shp_readiness_probe_params
+    - field.field.node.shp_project.field_shp_readiness_probe_port
+    - field.field.node.shp_project.field_shp_readiness_probe_type
```
Comment (singularo) [inline web/modules/custom/shepherd/shp_content_types/config/install/core.entity_form_display.node.shp_project.default.yml]: @pingers Do you think I should add in the timeout fields as well now, or just use sensible defaults

File: web/modules/custom/shepherd/shp_orchestration/src/OrchestrationProviderInterface.php [hunk 3]
```diff
@@ -139,12 +142,22 @@ public function createdEnvironment(
    *   Absolute url for the environment.
    * @param string $builder_image
    *   An s2i-enabled image to use to build (and run) the source code.
+   * @param string $domain
```
Comment (singularo) [inline web/modules/custom/shepherd/shp_orchestration/src/OrchestrationProviderInterface.php]: Fixing things that were missing here

**Comment by singularo** [comment]
Also requires https://github.com/universityofadelaide/openshift-client/pull/14

File: web/modules/custom/shepherd/shp_content_types/config/install/core.entity_form_display.node.shp_project.default.yml (hunk 2 above)
Comment (pingers) [reply web/modules/custom/shepherd/shp_content_types/config/install/core.entity_form_display.node.shp_project.default.yml]: Go with the sensible defaults and we'll add fields if needed.

File: web/modules/custom/shepherd/shp_orchestration/src/Service/Environment.php [hunk 4]
```diff
@@ -141,13 +153,15 @@ public function updated(NodeInterface $node) {
    * @return bool
    */
   public function deleted(NodeInterface $node) {
+    /** @var NodeInterface $site */
```
Comment (pingers) [inline web/modules/custom/shepherd/shp_orchestration/src/Service/Environment.php]: These need to be fully qualified namespaces.

**Comment by pingers** [comment]
:ship:

**Pull request merged**